File: api/src/main/java/it/futurecraft/sorrentino/database/enities/UserEntity.java
package it.futurecraft.sorrentino.database.enities;

import it.futurecraft.sorrentino.database.Entity;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

public interface UserEntity extends Entity<UUID> {
    @NotNull String displayName();

    void displayName(@NotNull String value);

    int twitchId();

    void twitchId(int value);

    @NotNull String twitchUsername();

    void twitchUsername(@NotNull String value);

    boolean streamer();

    void streamer(boolean value);
}

File: paper/src/main/kotlin/it/futurecraft/sorrentino/database/user/UserEntity.kt
package it.futurecraft.sorrentino.database.user

import it.futurecraft.sorrentino.database.User
import org.jetbrains.exposed.v1.core.dao.id.EntityID
import org.jetbrains.exposed.v1.dao.UUIDEntity
import org.jetbrains.exposed.v1.dao.UUIDEntityClass
import java.util.*

class UserEntity(id: EntityID<UUID>) : UUIDEntity(id), User {
    companion object : UUIDEntityClass<UserEntity>(UserTable)

    override var uuid: UUID = id.value

    override var twitchId by UserTable.twitchId

    override var twitchName by UserTable.twitchName

    override var displayName by UserTable.displayName

    override var streamer by UserTable.streamer
}
